README.md

```diff
@@ -9,6 +9,21 @@ rpm-ostree install greenboot greenboot-status greenboot-ostree-grub2
 systemctl reboot
 ```
 
+## How does it work?
+- `greenboot-rpm-ostree-grub2-check-fallback` service runs **before** `greenboot healthcheck` and checks whether the GRUB2 environment variable `boot_counter` is -1. 
+  - If it is -1, this would mean that the system is in a fallback deployment and would execute `rpm-ostree rollback` to go back to the previous, working deployment. 
+  - If `boot_counter` is not -1, nothing is done in this step.
+- `greenboot-healthcheck` service runs **before** systemd's [boot-complete.target](https://www.freedesktop.org/software/systemd/man/systemd.special.html#boot-complete.target). It launches `/usr/libexec/greenboot/greenboot check`, which runs the `required.d` and `wanted.d` scripts.
+  - If any script in the `required.d` folder fails, `redboot.target` is called.
+    - It triggers `redboot-task-runner` service, which launches `/usr/libexec/greenboot/greenboot red`. This will run the scripts in `red.d` folder.
```

Comment:
```suggestion
    - It triggers `redboot-task-runner.service`, which launches `/usr/libexec/greenboot/greenboot red`. This will run the scripts in `red.d` folder.
```